hiero-observability/hiero-metrics-core/src/main/java/module-info.java

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: Apache-2.0
+module org.hiero.metrics.core {
+    uses org.hiero.metrics.api.core.MetricsRegistrationProvider;
+    uses org.hiero.metrics.api.export.MetricsExporterFactory;
+
+    exports org.hiero.metrics.api;
+    exports org.hiero.metrics.api.core;
+    exports org.hiero.metrics.api.utils;
+    exports org.hiero.metrics.api.measurement;
+    exports org.hiero.metrics.api.export;
+    exports org.hiero.metrics.api.export.snapshot;
+    exports org.hiero.metrics.api.export.extension;
+    exports org.hiero.metrics.api.export.extension.writer;
+    exports org.hiero.metrics.api.stat;
+    exports org.hiero.metrics.api.stat.container;
+    exports org.hiero.metrics.internal.export.config to
+            com.swirlds.config.impl,
+            com.swirlds.config.extensions;
+
+    requires transitive com.swirlds.base;
+    requires java.management;
+    requires jdk.management;
+    requires jdk.sctp;
+    requires org.apache.logging.log4j;
+    requires static transitive com.swirlds.config.api;
+    requires static transitive com.github.spotbugs.annotations;
+    requires static transitive com.google.auto.service;
```

Comment:
Ideally this list would be much smaller
```suggestion
    requires static transitive com.swirlds.config.api;
    requires static transitive com.github.spotbugs.annotations;
```
I'm not certain about the config API. If it's not strictly necessary in the "core" module then it would be nice to remove that as well (particularly as it brings in some of the other dependencies we're removing).

Ideally the "core" module would not directly refer to the `JvmMetricsRegistration` class or the `MetricsExportManagerConfigurationExtension` class (those could, or should?, be in separate modules loaded at runtime and provided by deployment).